### PR TITLE
sd-bus: derive uid from cgroup if possible

### LINKED ISSUE #390: sd_bus_default doesn't work without kdbus
`sd_bus_default` doesn't seem to work: it's returning ENOENT

``` C
#include <systemd/sd-bus.h>
#include <stdio.h>
#include <string.h>

int main() {
    sd_bus *bus;
    int err = sd_bus_default(&bus);
    if (err < 0) {
        printf("%s\n", strerror(-err));
        return 1;
    }
    return 0;
}
```

```
$ gcc bus_test.c -lsystemd && sudo ./a.out
No such file or directory
```

strace output: http://sprunge.us/IfcS of note is that it fails straight after (and with the same error code as):

```
open("/sys/fs/kdbus/0-user/bus", O_RDWR|O_NOCTTY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
```

### PR DESCRIPTION
**Pull request by dvdhrm**
Whenever we run in a user context, sd_bus_{default_user,open_user}() and
friends should always connect to the user-bus of the current context,
instead of deriving the uid from getuid(). This allows us running
programs via sudo/su, without the nasty side-effect of accidentally
connecting to the root user-bus.

This patch enforces the idea of making su/sudo _not_ opening sessions by
default. That is, all they do is raising privileges, but keeping
everything set as before. You can still use su/sudo to open real sessions
by requesting a login-session (or loading pam_systemd otherwise).
However, in this case XDG_RUNTIME_DIR= will not be set (as usual in these
cases), hence, you will not be able to connect to _any_ user-bus.

Long story short: With this patch applied, both:
        - ./busctl --user
        - sudo ./busctl --user
..will successfully connect to the user-bus of the local user.

Fixes #390.

### CODE REVIEW

**Comment by dvdhrm** [comment]
@poettering, please review!

**Comment by poettering** [comment]
Looks good.
